File: csharp/0583两个字符串的删除操作.cs
public partial class Solution
{
    public int MinDistance(string word1, string word2)
    {
        int m = word1.Length + 1;
        int n = word2.Length + 1;

        int[,] dp = new int[m, n];

        for (int i = 1; i < m; i++)
        {
            dp[i, 0] = i;
        }
        for (int i = 1; i < n; i++)
        {
            dp[0, i] = i;
        }

        for (int i = 1; i < m; i++)
        {
            for (int j = 1; j < n; j++)
            {
                if(word1[i-1]==word2[j-1])
                    dp[i, j] = dp[i - 1, j - 1];
                else
                    dp[i, j] = Math.Min(dp[i - 1, j] + 1, Math.Min(dp[i, j - 1] + 1, dp[i - 1, j - 1] + 2));
            }
        }

        return dp[m - 1,n - 1];
    }
}

File: python/main/0583两个字符串的删除操作.py

def minDistance(word1: str, word2: str) -> int:
    m=len(word1)+1
    n=len(word2)+1

    dp=[[0]*n for _ in range(m)]

    # 初始化
    for i in range(1,m):
        dp[i][0]=i
    for i in range(1,n):
        dp[0][i]=i

    for i in range(1,m):
        for j in range(1,n):
            if(word1[i-1]==word2[j-1]):
                dp[i][j]=dp[i-1][j-1]
            else:
                dp[i][j]=min(dp[i-1][j]+1,dp[i][j-1]+1,dp[i-1][j-1]+2)
    
    return dp[-1][-1]